tensorflow/lite/kernels/add_n_test.cc

```diff
@@ -88,6 +88,77 @@ TEST(IntegerAddNOpModel, AddMultipleTensors) {
   EXPECT_THAT(m.GetOutput(), ElementsAreArray({-9, -1, 11, 11}));
 }
 
+class QuantizedAddNOpModel : public BaseAddNOpModel {
+ public:
+   using BaseAddNOpModel::BaseAddNOpModel;
+
+  template <typename integer_dtype>
+  std::vector<float> GetDequantizedOutput() {
+    return Dequantize<integer_dtype>(ExtractVector<integer_dtype>(output_),
+                                     GetScale(output_), GetZeroPoint(output_));
+  }
+};
+
+// for quantized Add, the error shouldn't exceed step
+float GetTolerance(float min, float max) {
+  float kQuantizedStep = (max - min) / 255.0;
+  return kQuantizedStep;
+}
+
+template <TensorType tensor_type, typename integer_dtype>
+void QuantizedTestsAddN() {
+  float kQuantizedTolerance = GetTolerance(-1.0, 1.0);
+
+  std::vector<std::vector<float>> inputs1 = {
+                                              {0.1, 0.1, 0.3, 0.4},
+                                              {-0.8, 0.2, 0.4, 0.7},
+                                              {-0.8, 0.2, 0.7, 0.3}
+                                             };
+  std::vector<std::vector<float>> inputs2 = {
+                                              {0.2, 0.4, 0.3, 0.1},
+                                              {0.6, 0.4, 0.2, -0.8},
+                                              {0.6, 0.4, -0.8, 0.5}
+                                             };
+  std::vector<std::vector<float>> inputs3 = {
+                                              {0.3, 0.2, 0.1, -0.1},
+                                              {0.2, 0.3, 0.1, 0.8},
+                                              {0.2, 0.3, 0.1, 0.1}
+                                             };
+
+  std::vector<std::vector<float>> results = {
+                                              {0.6, 0.7, 0.7, 0.4},
+                                              {0.0, 0.9, 0.7, 0.7},
+                                              {-0.0, 0.9, -0.0, 0.9}
+                                             };
+
+  for (size_t i = 0; i < inputs1.size(); ++i) {
+
+    QuantizedAddNOpModel m({{tensor_type, {1, 2, 2, 1}, -1.0, 1.0},
```

Comment:
Since rescaling is a critical feature in the code, can we use different scale for those tests?